components/lcd/esp_lcd_ssd1681/examples/epaper_lvgl_demo/main/main.c

```diff
@@ -0,0 +1,265 @@
+        .dc_gpio_num = EXAMPLE_PIN_NUM_EPD_DC,
+        .cs_gpio_num = EXAMPLE_PIN_NUM_EPD_CS,
+        .pclk_hz = EXAMPLE_LCD_PIXEL_CLOCK_HZ,
+        .lcd_cmd_bits = EXAMPLE_LCD_CMD_BITS,
+        .lcd_param_bits = EXAMPLE_LCD_PARAM_BITS,
+        .spi_mode = 0,
+        .trans_queue_depth = 10,
+        .on_color_trans_done = NULL,
+    };
+    // --- Attach the LCD to the SPI bus
+    ESP_ERROR_CHECK(esp_lcd_new_panel_io_spi((esp_lcd_spi_bus_handle_t) LCD_HOST, &io_config, &io_handle));
+    esp_lcd_panel_handle_t panel_handle = NULL;
+
+    // --- Create esp_lcd panel
+    esp_lcd_ssd1681_config_t epaper_ssd1681_config = {
+        .busy_gpio_num = EXAMPLE_PIN_NUM_EPD_BUSY,
+        .non_copy_mode = true,
+    };
+    esp_lcd_panel_dev_config_t panel_config = {
+        .reset_gpio_num = EXAMPLE_PIN_NUM_EPD_RST,
+        .flags.reset_active_high = false,
+        .vendor_config = &epaper_ssd1681_config
+    };
+    gpio_install_isr_service(0);
+    ESP_ERROR_CHECK(esp_lcd_new_panel_ssd1681(io_handle, &panel_config/*, &epaper_ssd1681_config*/, &panel_handle));
+    // --- Reset the display
+    ESP_LOGI(TAG, "Resetting e-Paper display...");
+    ESP_ERROR_CHECK(esp_lcd_panel_reset(panel_handle));
+    vTaskDelay(100 / portTICK_PERIOD_MS);
+    // --- Initialize panel
+    ESP_LOGI(TAG, "Initializing e-Paper display...");
+    ESP_ERROR_CHECK(esp_lcd_panel_init(panel_handle));
+    // --- Register the e-Paper refresh done callback
+    epaper_panel_callbacks_t cbs = {
+        .on_epaper_refresh_done = epaper_flush_ready_callback
+    };
+    epaper_panel_register_event_callbacks(panel_handle, &cbs, &disp_drv);
+    vTaskDelay(100 / portTICK_PERIOD_MS);
+    // --- Turn on display
+    ESP_LOGI(TAG, "Turning e-Paper display on...");
+    ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(panel_handle, true));
+    vTaskDelay(100 / portTICK_PERIOD_MS);
+    // --- Configurate the screen
+    // NOTE: the configurations below are all FALSE by default
+    ESP_ERROR_CHECK(esp_lcd_panel_mirror(panel_handle, false, false));
+    ESP_ERROR_CHECK(esp_lcd_panel_swap_xy(panel_handle, false));
+    // We need to invert color because we use 1 to represent WHITE
+    esp_lcd_panel_invert_color(panel_handle, false);
+    // NOTE: Calling esp_lcd_panel_disp_on_off(panel_handle, true) will reset the LUT to the panel built-in one,
+    // custom LUT will not take effect any more after calling esp_lcd_panel_disp_on_off(panel_handle, true)
+    ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(panel_handle, true));
+    // --- Initialize LVGL
+    ESP_LOGI(TAG, "Initialize LVGL library");
+    lv_init();
+    // alloc draw buffers used by LVGL
+    // it's recommended to choose the size of the draw buffer(s) to be at least 1/10 screen sized
+    lv_color_t *buf1 = heap_caps_malloc(EXAMPLE_LCD_H_RES * 200 * sizeof(lv_color_t), MALLOC_CAP_DMA);
+    assert(buf1);
+    lv_color_t *buf2 = heap_caps_malloc(EXAMPLE_LCD_H_RES * 200 * sizeof(lv_color_t), MALLOC_CAP_DMA);
```

Comment:
There can be used `EXAMPLE_LCD_H_RES` or `EXAMPLE_LCD_V_RES` instead of 200 in all cases.

Reply:
I think it's OK here, since we do not have to allocate a draw buffer that is exactly the size of the screen. For most cases, we allocate a buffer with H_RES * line_num.